### DISCUSSION TITLE
rx.redirect() to HTTP referer

### DISCUSSION THREAD

**Discussion by ijohn** [reflex-dev/reflex#3742] [Q&A] [answered]
Hi all!

I have multiple pages with `on_load=State.check_login`. I want the user to be able to bookmark the pages. Thus I need the authentication method to `rx.redirect()` to the referer. How do I do that? Thanks

**Comment by picklelo** [comment]
Hey! Have you checked out the [auth page](https://reflex.dev/docs/authentication/authentication-overview/)? If you look at the local auth example, it has a [method](https://github.com/reflex-dev/reflex-examples/blob/bdd651e1da0254bae41ea45f6dd86cac539b0794/twitter/twitter/state/base.py#L19) to redirect users who aren't logged in.

**Reply by ijohn** [reply, marked as answer]
Hi! Thanks for the response.

I had checked the docs and the method you pointed out before asking. That method doesn't answer my question.

For the benefit of others trying to figure out the same issue:

Lendemor (Legendmor) in Reflex's Discord helped me out with the following response:

"When running check_login, you can store the current page URL before going to /login/. 
Then when the login is complete, do a rx.redirect to the URL you stored previously."
